### ISSUE TITLE
Better Error Handling

### ISSUE THREAD

**Issue by maljovec** [idaholab/raven#221]
--------
Issue Description
--------
##### What did you expect to see happen?

A user attempted to use a malformed csv file (trailing empty lines) and the program should have exited with a message telling them that the input was malformed and how it was malformed (empty lines detected or something similar).

##### What did you see instead?

The program crashed with a non-intuitive error message something like:

```
TypeError: ufunc 'subtract' did not contain a loop with signature matching types dtype('<U21') dtype('<U21') dtype('<U21')
```

##### Do you have a suggested fix for the development team?

This should be handled when reading data from a csv.

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

[example.zip](https://github.com/idaholab/raven/files/1089740/example.zip)

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements? __Does not affect requirements__
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.  __Not a defect__
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

**Comment by maljovec** [comment]
This should be addressed in [maljdan/error_messages](https://github.com/idaholab/raven/tree/maljdan/error_messages)

**Comment by PaulTalbot-INL** [comment]
No additional notifications necessary; approved to be closed by #223.
